src/main/services/device.ts

```diff
@@ -65,7 +65,7 @@ export class DeviceIdManager implements IDeviceIDManager {
             const deviceConfig: DeviceConfig = {
                 device_id: deviceId,
             };
-            this.saveDeviceConfigJson(deviceConfig);
+            await this.saveDeviceConfigJson(deviceConfig);
```

Comment:
the promise is not returned from the function https://github.com/edrlab/thorium-reader/blob/8069ae9d3b6c52036304d79cb35bb0886428031e/src/main/services/device.ts#L107